jobs/update_orphaning_dashboard_etl.py

```diff
@@ -0,0 +1,1236 @@
+        destination_uri,
+        location="US",
+        job_config=job_config
+    )  
+    extract_job.result()  # Waits for job to complete.
+
+    print(
+        "Exported {} to {}".format(table_ref, destination_uri)
+    )
+
+    longitudinal_shim_raw_from_avro=spark.read.format("avro").load(GCS_TABLE_DUMP_PATH)
+
+    def flatten_settings_update(settings):
+        channel_list = list(map(lambda x: x[0], settings["update"]["channel"]))
+        enabled_list = list(map(lambda x: x[0], settings["update"]["enabled"]))
+        return {"update": {"channel": channel_list, "enabled": enabled_list}}
+
+    return_struct_type = StructType([StructField("update", StructType([StructField("channel", ArrayType(StringType())), StructField("enabled", ArrayType(BooleanType()))]))])
+    udf_flatten_settings_update = udf(flatten_settings_update, return_struct_type)
+
+    def fix_schema(raw_longitudinal_df):
+        with_settings_fixed = raw_longitudinal_df.withColumn("settings_fixed", udf_flatten_settings_update("settings")).drop("settings").withColumnRenamed("settings_fixed", "settings")
+        return with_settings_fixed
+
+
+
+    longitudinal_shim_raw_from_avro_fixed = fix_schema(longitudinal_shim_raw_from_avro)
+
+
+
+
+    def merge_keyed_count_histograms(histograms, histogram_name):
+        keys = []
+        for histogram_struct in histograms:
+            histogram_array = histogram_struct[histogram_name]
+            if histogram_array:
+                for key, count_histogram_string in histogram_array:
+                    if key not in keys:
+                        keys.append(key)
+
+        res = {}
+        n_hist = len(histograms)
+        for key in keys:
+            res[key] = [0]*n_hist
+            for i, histogram_struct in enumerate(histograms):
+                histogram_array = histogram_struct[histogram_name]
+                if histogram_array:
+                    for hist_key, count_histogram_string in histogram_array:
+                        if hist_key == key:
+                            res[key][i]=json.loads(count_histogram_string)['values'].get('0', 0)
+
+        return res
+
+    merge_keyed_count_histograms_udf = F.udf(merge_keyed_count_histograms, MapType(StringType(), ArrayType(IntegerType())))
+
+    def merge_keyed_count_histogram_col(df, col_name):
+        return df.withColumn(col_name+"_merged", merge_keyed_count_histograms_udf(col_name, F.lit(col_name))).drop(col_name).withColumnRenamed(col_name+"_merged", col_name)
+
+    def merge_enumerated_histograms(histograms, histogram_name, n_values):
+        res = []
+        all_null = True # needed to maintain compatibility with Longitudinal
+        for histogram_string_struct in histograms:
+            compacted_histogram = [0]*(int(n_values)+1)
+            histogram_string = histogram_string_struct[histogram_name]
+            if histogram_string:
+                all_null = False
+                values = json.loads(histogram_string)['values']
```

Comment:
This strikes me as a likely place of failure for a histogram that is missing `values`. Consider using `.get`